src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -671,17 +679,19 @@
 
 (pls/defn-validated update-catalog-associations!
   "Adds/updates/deletes the edges and resources for the given certname"
-  [catalog-id :- Number
+  [certname-id :- Number
```

Comment:
It's rather strange that we're typing these as java.lang.Number. 

```The abstract class Number is the superclass of classes BigDecimal, BigInteger, Byte, Double, Float, Integer, Long, and Short.```

I think Long is closer to what we want.